### ISSUE TITLE
Make sub items in navigation block collapsable

### ISSUE THREAD

**Issue by tomxygen** [WordPress/gutenberg#62172]
## What problem does this address?
Currently, when adding sub items in the navigation block, they're collapsed on desktop and can be revealed by hovering over the parent item.
On mobile however, they're always shown. 

I am building a WooCommerce store and I'd like to place all the product categories as sub-items of the Catalog parent item.
On mobile this is the result:
![IMG_9904](https://github.com/WordPress/gutenberg/assets/47855768/22043162-051f-49c9-9236-d338471883cf)

## What is your proposed solution?

Sub-items should be collapsable.

**Comment by phanduynam** [comment]
It's been a long time and no one has solved this problem

**Comment by hagege** [comment]
It would indeed be wonderful if this feature were realized.

**Comment by JosVelasco** [comment]
I agree. The mobile version shouldn't feel like a simplified version.

**Comment by paaljoachim** [comment]
I will add in a few navigation mobile issues:

Improved mobile navigation layout
https://github.com/WordPress/gutenberg/issues/58036

Here is a general search with the Navigation label and search for mobile:
https://github.com/WordPress/gutenberg/issues?q=is%3Aissue%20state%3Aopen%20mobile%20label%3A%22%5BBlock%5D%20Navigation%22%20

**Comment by phanduynam** [comment]
When will this problem be solved?
